src/__tests__/app.test.ts

```diff
@@ -1,29 +1,37 @@
+// NODE_ENV *before* you require app.js or it will use the wrong (global) NODE_ENV
+// tslint:disable:no-let
+let originalNodeEnv: string;
+originalNodeEnv = process.env.NODE_ENV || "dev";
+process.env.NODE_ENV = "production";
```

Comment:
I'm not sure about this approach, as it may cause problems when tests are run in parallel, my suggestion would be to avoid having functions reading directly from the environment - instead the "environment" should be passed as parameter - in general functions should try hard to be pure

Reply:
for instance, in this case, this case, the `app` object shouldn't be implicitly created by just including `../app` - instead, the `app` module should export a function that takes an environment enum (e.g. `DEVELOPMENT, PROD`) and returns an `app` object

Reply:
this approach will enable testing the behavior of `app` in isolation